tests/test_reader.py

```diff
@@ -0,0 +1,105 @@
+import matplotlib_agg  # noqa: F401
+import os
+import sys
+import pytest
+import numpy as np
+from numpy.testing import assert_array_equal, assert_array_almost_equal
+import matplotlib.pyplot as plt
+from anesthetic import MCMCSamples, NestedSamples
+try:
+    import getdist
+    import montepython  # noqa: F401
+except ImportError:
+    pass
+
+
+def test_read_getdist():
+    np.random.seed(3)
+    mcmc = MCMCSamples(root='./tests/example_data/gd')
+    mcmc.plot_2d(['x0', 'x1', 'x2', 'x3'])
+    mcmc.plot_1d(['x0', 'x1', 'x2', 'x3'])
+
+    mcmc = MCMCSamples(root='./tests/example_data/gd_single')
+    mcmc.plot_2d(['x0', 'x1', 'x2', 'x3'])
+    mcmc.plot_1d(['x0', 'x1', 'x2', 'x3'])
+    plt.close("all")
+
+
+@pytest.mark.xfail('getdist' not in sys.modules,
+                   raises=NameError,
+                   reason="requires getdist package")
+def test_read_cobayamcmc():
+    np.random.seed(3)
+    mcmc = MCMCSamples(root='./tests/example_data/cb')
+    mcmc.plot_2d(['x0', 'x1'])
+    mcmc.plot_1d(['x0', 'x1'])
+    plt.close("all")
+
+    # compare directly with getdist
+    mcmc_gd = getdist.loadMCSamples(file_root="./tests/example_data/cb")
+    assert_array_almost_equal(mcmc.logL, mcmc_gd.loglikes, decimal=15)
```

Comment:
This type of tests works with the new `cobayamcmcreader` and the `cb` example data, but it does _not_ currently work with the `getdistreader` and the `gd` example data.

Reply:
fixed with new example data